src/Form.tsx

```diff
@@ -2,57 +2,59 @@ import './Form.scss'
 import CheckboxInput from './inputs/Checkbox'
 import TogglableColorPicker from './inputs/TogglableColorPicker'
 import IntegerInput from './inputs/Integer'
-import { StitchPattern, Color, ColorSequenceArray } from './types'
+import { Color, SwatchConfig } from './types'
 import { getRandomNotWhiteColor, totalColorSequenceLength } from './color'
 
-type SwatchConfigurationData = {
-  colorSequence: ColorSequenceArray,
-  stitchesPerRow: number,
-  stitchPattern: StitchPattern,
-  numberOfRows: number,
-  colorShift: number,
-  staggerLengths: boolean,
+type DisplayData = {
   showRowNumbers: boolean
 }
 
-type FormValue = keyof(SwatchConfigurationData)
+type FormValue = keyof(SwatchConfig) | keyof(DisplayData)
 
 const Form = (
-  { formData, setFormData } : 
+  { swatchData, setSwatchData, displayData, setDisplayData } :
   {
-    formData: SwatchConfigurationData,
-    setFormData: (data: SwatchConfigurationData) => void
+    swatchData: SwatchConfig,
+    setSwatchData: (data: SwatchConfig) => void,
+    displayData: DisplayData,
+    setDisplayData: (data: DisplayData) => void
   }
 ) => {
 
-  const { colorSequence, stitchesPerRow, numberOfRows, colorShift, staggerLengths, stitchPattern, showRowNumbers } = formData;
+  const { colorSequence, stitchesPerRow, numberOfRows, colorShift, staggerLengths, stitchPattern } = swatchData;
+  const { showRowNumbers } = displayData;
 
   const setFormValue = (name: FormValue, value : string | number | boolean) => {
```

Comment:
This is the main weird thing, I look to see what object the FormValue is in instead of breaking it into two different methods